libgnome-desktop/gnome-wall-clock.c

```diff
@@ -288,27 +292,44 @@ update_clock (gpointer data)
 			   in 24-hour mode. */
 			format_string = show_seconds ? _("%R:%S") : _("%R");
 		}
+		format_string_dup = g_strdup (format_string);
 	} else {
+		GDateTime *now;
+		gchar *ampm_string;
+
 		if (show_full_date) {
 			/* Translators: This is a time format with full date used
 			   for AM/PM. */
-			format_string = show_seconds ? _("%a %b %e, %l:%M:%S %p")
-				: _("%a %b %e, %l:%M %p");
+			format_string = show_seconds ? _("%a %b %e, %l:%M:%S")
```

Comment:
Same comments of the control center PR apply here. Please use new translatable strings for the fallback AM/PM instead of changing all these.